Add navigation documentation for previous/next page navigation

xSaints19x opened the pull request:
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1975

**Anything you'd like to highlight/discuss:**
I implemented the 1st solution suggested by @tlylt in the #1975 issue where I documented a common way to add page navigation to a project quickly. Documentation is broken down into 2 parts namely, Adding the required file and folder and Editing the content files, where code examples have been added for clarity.

**Testing instructions:**
Navigate to the `Adding Navigation Buttons` subsection under the `Working with sites` section to verify the documentation: https://deploy-preview-2048--markbind-master.netlify.app/userguide/addingnavigationbuttons

**Proposed commit message: (wrap lines at 72 characters)**
Add documentation for previous/next page navigation

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [ ] No unrelated changes

tlylt commented on docs/userGuide/addingNavigationButtons.md:
````diff
@@ -0,0 +1,157 @@
+
+You should now be able to navigate between section `Topic 1` and `Topic 2` with ease.
+
+<box type="tip">
+
+Whenever you create a new page, be sure to include the following code below and replace the parts enclosed in the square brackets e.g `[Name_Of_Section]`   
+
+{% raw %}
+```html
+<span id="title" class="d-none">[Name_Of_Section]</span>
+
+{% from "njk/common.njk" import previous_next %}
+{{ previous_next('[Previous_Page_Filename]', '[Next_Page_Filename]') }}
+```
+{% endraw %}
+
+</box>
````
Could we move expand on this and move this up before the examples to give readers an understanding of what the code is doing? Mentioning the parts that readers need to change as well as which parts of the UI are affected will give them a better idea of what to do when customizing for their own website.

Also because this whole feature involves njk syntax that readers may not be aware of, we could link some relevant resources, and be a bit more explicit when mentioning what we are doing in the code, like defining an invisible title fragment that's used in the common.njk (same goes for the other important concepts)

xSaints19x replied:
Oh yes, I will work on that but I have a question regarding how I should go about explaining the idea behind what the code does. I thought of writing it down as a markdown comment above the different parts that require explanation, not sure if you think that's advisable? E.g `defining an invisible title fragment that's used in the common.njk` would be placed above the `<span id="title" class="d-none">[Name_Of_Section]</span>` in the form of a comment

tlylt replied:
Hi @xSaints19x, thank you for moving the tip up. I think it is clearer now but would be great if we can improve it further. Here are my suggestions in more greater detail, please let me know if it works for you:
1. Adding an overview of the whole process here before listing out the steps. What we can add here is a brief introduction as to how the entire thing works. So perhaps along the lines of "we will be making use of nunjucks templates (link to https://markbind.org/userGuide/markBindSyntaxOverview.html#support-for-nunjucks) .... and it works by defining xxx...)
![image](https://user-images.githubusercontent.com/41845017/208028671-fd7b95ff-8d15-440b-bec0-2813d32f9ed7.png)
2. Refactor the paragraph
![image](https://user-images.githubusercontent.com/41845017/208029351-8c58d08c-4f41-4ab9-aac8-9230dcb56986.png)
- it should not be a tip or closable box and the information is quite essential to understanding the code snippet
- avoid using `refer to explanation point 3 below -->`, if an author copies this code snippet, he/she has to painstakely remove the comments that are not appropriate when appearing in their code file. Would be great if the comments are self-sufficient.
- minor edits on capitalization "Effects after changes ..." and not "Effects after Changes..."
- make use of the line number feature and address the line number in your explanation instead of the more verbose "that line of code ...."

tlylt replied:
Sounds good 👍 Let's try that

tlylt commented on docs/userGuide/addingNavigationButtons.md:
````diff
@@ -0,0 +1,157 @@
+{% endraw %}
+
+{% raw %}
+```html {heading="**topic2.md**" .line-numbers}
````
```suggestion
```html {heading="**topic2.md**" .line-numbers highlight-lines="1,8-9"}
```

tlylt commented on docs/userGuide/addingNavigationButtons.md:
````diff
@@ -0,0 +1,157 @@
+<div class="indented">
+
+{% raw %}
+```html {heading="**topic1.md**" .line-numbers}
````
```suggestion
```html {heading="**topic1.md**" .line-numbers highlight-lines="1,9-10"}
```

tlylt reviewed (changes requested):
Hi @xSaints19x, thank you for the PR, great work on the examples!

I have followed and tested the instruction and it works as expected. Left some comments on the docs, mainly for
- more explicit explanation of what the snippet is doing because some variables/concepts have to be understood before users can use it
- minor improvements to utilize MarkBind features such as tree diagram and line highlighting
- miscellaneous edits

xSaints19x commented on docs/userGuide/addingNavigationButtons.md:
```diff
@@ -0,0 +1,157 @@
+
+<div class="lead" id="overview">
+
+Navigation buttons can **greatly help your visitors of your website browse through your contents with ease**.  
```
Hi Yong Liang thanks for the suggestion here, the wording `visitors of your website` was actually taken from `navigation.md` file which is a subsection in the `Using Components` section. I thought I wanted to standardise the wording so not sure if I should change that

tlylt replied:
I see, should be fine either way since this wording is trivial.

tlylt replied:
Hi @xSaints19x could you remove the trailing space?

tlylt commented on docs/userGuide/addingNavigationButtons.md:
````diff
@@ -0,0 +1,157 @@
+<div class="indented">
+
+{% raw %}
+```html
````
@xSaints19x are you ok with this suggestion?

xSaints19x replied:
Yeap, have committed your suggestion to the PR!

tlylt commented:
The `packages/core/package-lock.json` file seems to be an unwanted change, will need to revert it.

xSaints19x commented:
> The `packages/core/package-lock.json` file seems to be an unwanted change, will need to revert it.

I've reverted the commit and removed the package-lock.json file in the latest commit

tlylt commented:
Hi @xSaints19x, thanks for making the changes (almost there now! 🦾 )! Have you taken a look at the rendered [output](https://deploy-preview-2048--markbind-master.netlify.app/userguide/addingnavigationbuttons)? Good to review it from a user's POV.

Some things that I observed:
- the Intra-site links are directed to the absolute path of deployed markbind site, instead of relative to the current docs (I previously used the absolute link in the comment (https://github.com/MarkBind/markbind/pull/2048#discussion_r1050387559) as an example, not meant to mean it should link directly there, you can take a look at other pages in the docs to see how to reference another page correctly)
- seems like there's a duplication of content
- here's my suggestions on the overview

"
Overview of steps

1. Define a function to read the page title and create navigation buttons via Nunjucks (link)

2. Define a title variable in the content file and invoke the function created to render the navigation buttons

In the following example, we will be editing the site content generated from the default template(link) (by running `markbind init`)
"

What do you think?

xSaints19x commented:
Hi Yongliang,

Thanks for the suggestion I've amended the intra-site links and changed the overview steps with the one you recommended with just a minor addition of `in the terminal` for the part where you mentioned running the command `markbind init` to be clearer to the user.

tlylt commented on docs/userGuide/addingNavigationButtons.md:
```diff
@@ -0,0 +1,190 @@
+
+**Overview of steps**
+
+> 1. Define a function to read the page title and create navigation buttons via [Nunjucks](/userGuide/markBindSyntaxOverview.html#support-for-nunjucks)
+> 2. Define a title variable in the content file and invoke the function created to render the navigation buttons
```
```suggestion
1. Define a function to read the page title and create navigation buttons via [Nunjucks](/userGuide/markBindSyntaxOverview.html#support-for-nunjucks)
1. Define a title variable in the content file and invoke the function created to render the navigation buttons
```

xSaints19x replied:
Changed from quotation to points in latest commit!

tlylt commented on docs/userGuide/addingNavigationButtons.md:
```diff
@@ -0,0 +1,190 @@
+  _markbind
+  _site
+  contents
+  njk
```
```suggestion
  njk/
```

tlylt commented on docs/userGuide/addingNavigationButtons.md:
```diff
@@ -0,0 +1,190 @@
+
+<box type="tip">
+
+Whenever you create a new page, be sure to include the following code below and replace the parts enclosed in the square brackets e.g `[Name_Of_Section]`
```
Content here and line 176 is the same, is this intended?

tlylt reviewed (changes requested):
last few nits

tlylt reviewed (approved):
Thanks @xSaints19x, LGTM!

tlylt commented:
@all-contributors please add @xSaints19x for docs
